zink/codegen/src/contract.rs

```diff
@@ -0,0 +1,275 @@
+//! Derive macro for contract storage
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, Fields, ItemStruct, Type};
+
+/// Parse the input struct and generate the contract storage implementation
+pub fn parse(input: ItemStruct) -> TokenStream {
```

Comment:
instead a single `parse` function, I'd recommend to use a struct for making this more gracefully, for example, the implementation for the storage macro https://github.com/zink-lang/zink/blob/main/zink/codegen/src/storage.rs#L26